src/main/webapp/app/exam/participate/exam-navigation-bar/exam-navigation-bar.component.ts

```diff
@@ -45,6 +54,32 @@ export class ExamNavigationBarComponent implements OnInit {
                 this.itemsVisiblePerSide = 0;
             }
         });
+
+        const timingEntries = window.performance.getEntriesByType('navigation') as [PerformanceNavigationTiming];
+        const isReload = timingEntries.some((entry) => entry.name === window.location.href && entry.type === 'reload');
+        if (!isReload) {
```

Comment:
what happens if the user navigates into the exam from the starting page after having produced the unsynced state? This is not a "real reload" then, but rather a new second entry into the exam.

I think we have to use another mechanism. One idea that just came into my mind: Each time the user starts the exam, an exam session will be created (no matter if the user reloads the page or not). We could use this information and send one additional (transient) attribute to the client whether the user has started the exam for the first time or not based on the number of exam sessions in the database.

If the user has started the exam for the first time, we don't need to execute the code below. Otherwise, we execute the code below

Reply:
I think the server filters all "old" exam sessions and only attaches the newest one, which is something that we should not change, so I prefer a transient attribute